src/ir/module.cc

```diff
@@ -247,8 +227,8 @@ void IRModuleNode::AddTypeDef(const GlobalTypeVar& var, const TypeData& type, bo
   AddTypeDefUnchecked(var, type, update);
   // need to kind check at the end because the check can look up
   // a definition potentially
-  CHECK(relay::KindCheck(type, GetRef<IRModule>(this)) == TypeKind::kTypeData)
-      << "Invalid or malformed typedata given to module: " << type;
+  // CHECK(relay::KindCheck(type, GetRef<IRModule>(this)) == TypeKind::kTypeData)
+  //     << "Invalid or malformed typedata given to module: " << type;
```

Comment:
uncomment or remove.

Reply:
I opened a tracking issue for the kind checking issues, and did some of the refactors on this branch.